### PR TITLE
port:qca* : proposed adoption and introduction of Qt5 subports

### PR DESCRIPTION
**Pull request by RJVB**
The QCA ports have been without maintainer for a bit now, and lack the
possibility to install for Qt4 and Qt5 concurrently. That possibility is
essential for the future KF5 ports, so I propose that Marko and I adopt
this port and introduce the required changes. The ports as defined by
the current Portfile have seen thorough testing by both of us as well as
by a few "beta testers" of my KF5 ports.
I don't think there is any point in maintaining the Qt4 and Qt5 variants.
The current version also introduces a few changes that are unrelated to
Qt5/KF5 but aim to simplify dependencies by reducing the number of
subports. This is open for discussion w.r.t. the Qt4 versions. I don't
think there is much justification (for use with MacPorts) to build QCA
as a framework, but this could be reintroduced as a variant.

If the current version is really unacceptable for some reason we can also
resubmit it in a version catering only to Qt5.

### CODE REVIEW

**Comment by Veence** [comment]
The Qt4 variant, together with qca-ossl is required by QGis.

As for now, QGis still depends on Qt.4, though plans to release an overhauled version that would move towards Qt.5 are well under way (but still not finished). As long as that task is not completed, keeping a Qt.4 version is mandatory.

**Comment by RJVB** [comment]
Relax, it was never my intention to retire Qt4 support because it is also required for KDE4 PIM (I'm not yet planning a KF5 PIM implementation).

In fact, you're giving an additional argument why QCA *must* be co-installable in Qt4 and Qt5 versions.

Would it be a lot of work for you to test the proposed new version? I'm not claiming it'll be a drop-in replacement, but I'm not foreseeing any issues other than an imposed rebuild (probably an incremental one that's not much more than a re-configure and a re-link step for you ;)).

**Comment by Veence** [comment]
Oh, it’s okay, I wasn’t freaking out :)
No problem to check out your new version. Just pass it along, it’ll be my pleasure.

**Comment by mkae** [comment]
@Veence, have you been able to test this port for your use case? Would be good to know in order to move forward with this.

**Comment by mkae** [comment]
Successfully installed `qca-qt5` on a `qt5`-based install on an "El Capitan"-VM.

File: devel/qca/Portfile [hunk 1]
```diff
@@ -1,223 +1,187 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
-# $Id$
 
 PortSystem          1.0
-PortGroup           cmake 1.0
-PortGroup           active_variants 1.1
 
 name                qca
-version             2.1.0
-revision            3
```
Comment (mkae) [inline devel/qca/Portfile]: This is revision 3 and it should be kept further down for the old `qt4`-based port, no?!

File: devel/qca/Portfile [hunk 2]
```diff
@@ -1,223 +1,187 @@
...
+    checksums       rmd160  cc4040d013170ca5f21fbcefd0607daa8191d333 \
+                    sha256  ab0170696a8ff87588d38a72d36d0b4b42db9080c991d1a40ceb3f8311d289f9
+    use_xz          yes
+    distname        "qca-qt5-${version}"
+    livecheck.regex "qca-qt5-(\\d+(?:\\.\\d+)*)"
+} else {
+    version         2.1.0
+    set branch      [join [lrange [split ${version} .] 0 0] .]
+    PortGroup       qt4 1.0
+    set Qt_Major    4
```
Comment (mkae) [inline devel/qca/Portfile]: ... here we should keep revision set to 3.

**Comment by RJVB** [comment]
No, this is a bit confusing; the Qt4 version is 2.1.0, port revision 3, while the Qt5 version is 2.1.0.3, port revision 0.

**Comment by mkae** [comment]
That's the point for qt4, the revision is skipped in this PR and needs to be resurrected.

File: devel/qca/Portfile [hunk 3]
```diff
@@ -1,223 +1,188 @@
...
+    set Qt_Major    5
+    master_sites    http://download.kde.org/stable/qca-qt5/${version}/src
+    checksums       rmd160  cc4040d013170ca5f21fbcefd0607daa8191d333 \
+                    sha256  ab0170696a8ff87588d38a72d36d0b4b42db9080c991d1a40ceb3f8311d289f9
+    use_xz          yes
+    distname        "qca-qt5-${version}"
+    livecheck.regex "qca-qt5-(\\d+(?:\\.\\d+)*)"
+} else {
+    version         2.1.0
+    revision        3
```
Comment (mkae) [inline devel/qca/Portfile]: Keeping the revision at 3 should be fine. We don't need to revbump to 4, right?

File: devel/qca/Portfile (hunk 1 above)
Comment (RJVB) [inline devel/qca/Portfile]: As a matter of fact, bumping to 4 could be a good idea.

**Pull request merged**